### PR TITLE
Add container download and docker command

### LINKED ISSUE #9: readme instructions don't work
following the readme, if one installs the library and runs a predict model it says sif file not found. the sif file or docker image must exist somewhere so the user does not have to build it.

### PR DESCRIPTION
**Pull request by Hoda1394**
fixed #9 
There is an issue with the docker run that is related to the docker container but singularity works fine. I will create a separate issue for that.

### CODE REVIEW

**Comment by Hoda1394** [comment]
@satra - Ready to merge.

File: nobrainerzoo/cli.py [hunk 1]
```diff
@@ -110,20 +110,36 @@ def predict(
...
     
-    download_image = Path(__file__).resolve().parents[0] / "env/nobrainer-zoo_test.sif"
-    cmd0 = ["singularity", "run", download_image, "python3", "nobrainerzoo/download.py", model]
+    if container_type == "singularity":
+        
+        download_image = Path(__file__).resolve().parents[0] / "env/nobrainer-zoo_test.sif"
+        if not download_image.exists():
+            dwnld_cmd = ["singularity", "pull", "--dir", 
+                       str(Path(__file__).resolve().parents[0]/ "env"),
+                       "docker://hodadock/nobrainer-zoo:test"]
```
Comment (satra) [inline nobrainerzoo/cli.py:130]: shouldn't these be in the neuronets organization?

File: nobrainerzoo/cli.py [hunk 2]
```diff
@@ -110,20 +110,36 @@ def predict(
...
+            p = sp.run(dwnld_cmd, stdout=sp.PIPE, stderr=sp.STDOUT, text=True)
+            print(p.stdout)
+        cmd0 = ["singularity", "run", download_image, "python3", "nobrainerzoo/download.py", model]
+    elif container_type == "docker":
+        path = str(Path(__file__).resolve())+":"+str(Path(__file__).resolve())
+        file = str(Path(__file__).resolve() / "download.py")
+        # check output option
+        cmd0 = ["docker", "run","-v",path,"--rm","hodadock/nobrainer-zoo:test",
```
Comment (satra) [inline nobrainerzoo/cli.py:138]: and this?

Comment (Hoda1394) [reply nobrainerzoo/cli.py:138]: I considered a folder for all container images, they can be decentralized to organizations too. The only reason was the neurons image that is used for general purposes, so I do not need to include git-annex and Datalad to other images.

Comment (satra) [reply nobrainerzoo/cli.py:138]: simply saying to move this image to the neuronets organization instead of your personal account.

Comment (Hoda1394) [reply nobrainerzoo/cli.py:138]: Ah....ok, I thought you are talking about the structure.

File: nobrainerzoo/cli.py [hunk 3]
```diff
@@ -110,20 +110,36 @@ def predict(
...
+            p = sp.run(dwnld_cmd, stdout=sp.PIPE, stderr=sp.STDOUT, text=True)
+            print(p.stdout)
+        cmd0 = ["singularity", "run", download_image, "python3", "nobrainerzoo/download.py", model]
+    elif container_type == "docker":
+        path = str(Path(__file__).resolve())+":"+str(Path(__file__).resolve())
+        file = str(Path(__file__).resolve() / "download.py")
+        # check output option
+        cmd0 = ["docker", "run","-v",path,"--rm","neuronets/nobrainer-zoo:nobrainer", 
+                "python3", file, model]
+
```
Comment (kaczmarj) [inline nobrainerzoo/cli.py:140]: could you raise an error if the container_type is not docker or singularity?

```suggestion
        else:
            raise ValueError(f"unknown container type: {container_type}")
```

File: nobrainerzoo/cli.py [hunk 4]
```diff
@@ -178,12 +195,25 @@ def predict(
...
+        cmd_options = ["--nv", "-B", bind_paths, "-B", f"{out_path}:/output", "-W", "/output"]
+        cmd = ["singularity", "run"] + cmd_options + [image] + model_cmd.split() \
+            + model_options
+    elif container_type == "docker":
+        bind_paths_docker = []
+        for el in bind_paths:
+            bind_paths_docker += ["-v", f"{el}:{el}"]
+        cmd_options = bind_paths_docker + ["-v", f"{out_path}:/output", "-w", "/output", "--rm"]
+        cmd = ["docker", "run"] + cmd_options + [image] + model_cmd.split() \
+            + model_options
```
Comment (kaczmarj) [inline nobrainerzoo/cli.py:211]: could you raise an error if the container_type is not docker or singularity?

```suggestion
        else:
            raise ValueError(f"unknown container type: {container_type}")
```

**Pull request merged**